File: tests/fixtures/def-tree-complex.go
package sample

func entry() string {
	raw := loadConfig()
	parsed := parseConfig(raw)
	normalized := normalizeConfig(parsed)
	score := computeScore(normalized, 2)
	result := finalizeResult(score)
	audit(result)
	audit(result)
	return result
}

func loadConfig() string {
	return readEnv()
}

func readEnv() string {
	return sanitizeEnv("APP_MODE")
}

func sanitizeEnv(key string) string {
	return key
}

func parseConfig(raw string) string {
	if raw == "" {
		return defaultConfig()
	}
	return expandConfig(raw)
}

func defaultConfig() string {
	return "default"
}

func expandConfig(raw string) string {
	return raw + ":" + raw
}

func normalizeConfig(cfg string) string {
	compact := shrinkConfig(cfg)
	return compact
}

func shrinkConfig(cfg string) string {
	return cfg
}

func computeScore(cfg string, weight int) int {
	base := evaluate(cfg)
	weighted := applyWeight(base, weight)
	return combineScores(base, weighted)
}

func evaluate(cfg string) int {
	a := metricA(cfg)
	b := metricB(cfg)
	return add(a, b)
}

func metricA(cfg string) int {
	return lengthOf(cfg)
}

func metricB(cfg string) int {
	return lengthOf(cfg)
}

func lengthOf(cfg string) int {
	return len(cfg)
}

func applyWeight(value int, weight int) int {
	return multiply(value, weight)
}

func multiply(a int, b int) int {
	return a * b
}

func combineScores(a int, b int) int {
	return add(a, b)
}

func add(a int, b int) int {
	return a + b
}

func finalizeResult(value int) string {
	return formatResult(value)
}

func formatResult(value int) string {
	return "score:"
}

func audit(result string) string {
	return logResult(result)
}

func logResult(result string) string {
	return result
}

File: tests/fixtures/def-tree-complex.ts
export function entry() {
  const raw = loadConfig();
  const parsed = parseConfig(raw);
  const normalized = normalizeConfig(parsed);
  const score = computeScore(normalized, 2);
  const result = finalizeResult(score);
  audit(result);
  audit(result);
  return result;
}

export function loadConfig() {
  return readEnv();
}

export function readEnv() {
  return sanitizeEnv("APP_MODE");
}

export function sanitizeEnv(key: string) {
  return key.trim().toLowerCase();
}

export function parseConfig(raw: string) {
  if (!raw) {
    return defaultConfig();
  }
  return expandConfig(raw);
}

export function defaultConfig() {
  return "default";
}

export function expandConfig(raw: string) {
  return raw + ":" + raw;
}

export function normalizeConfig(cfg: string) {
  const compact = shrinkConfig(cfg);
  return compact;
}

export function shrinkConfig(cfg: string) {
  return cfg.replace("::", ":");
}

export function computeScore(cfg: string, weight: number) {
  const base = evaluate(cfg);
  const weighted = applyWeight(base, weight);
  return combineScores(base, weighted);
}

export function evaluate(cfg: string) {
  const a = metricA(cfg);
  const b = metricB(cfg);
  return add(a, b);
}

export function metricA(cfg: string) {
  return lengthOf(cfg);
}

export function metricB(cfg: string) {
  return lengthOf(cfg);
}

export function lengthOf(cfg: string) {
  return cfg.length;
}

export function applyWeight(value: number, weight: number) {
  return multiply(value, weight);
}

export function multiply(a: number, b: number) {
  return a * b;
}

export function combineScores(a: number, b: number) {
  return add(a, b);
}

export function add(a: number, b: number) {
  return a + b;
}

export function finalizeResult(value: number) {
  return formatResult(value);
}

export function formatResult(value: number) {
  return "score:" + value;
}

export function audit(result: string) {
  return logResult(result);
}

export function logResult(result: string) {
  return result;
}
